src/sqlfluff/core/rules/reference.py

```diff
@@ -0,0 +1,25 @@
+"""Components for working with object and table references."""
+from typing import Sequence, Tuple
+
+
+def object_ref_matches_table(
```

Comment:
My thought process for adding this function was to write this algorithm in such a way that it was independent of other parts of SQLFluff (parser, `AliasInfo` class, etc.) This makes it easier to unit test, but importantly, I think it's also easier to understand in its pure form using **only** built-in Python types.

Does this seem like a good decision decision? Is this an appropriate home for this code?

Reply:
Yeah I'm happy with keeping this separate, makes our logic for determining ref matches self contained and well defined which is useful for rule writers 👍